File: examples/injectable-twitter-server/scala/src/test/scala/com/twitter/finatra/example/TestQueue.scala
package com.twitter.finatra.example

import java.util.concurrent.atomic.AtomicInteger
import javax.inject.Singleton

@Singleton
class TestQueue extends Queue(Integer.MAX_VALUE) {
  val addCounter: AtomicInteger = new AtomicInteger(0)

  override def add(value: String): Boolean = {
    addCounter.incrementAndGet()
    super.add(value)
  }
}

File: examples/injectable-twitter-server/java/src/test/java/com/twitter/finatra/example/TestQueue.java
package com.twitter.finatra.example;

import java.util.concurrent.CountDownLatch;
import java.util.concurrent.atomic.AtomicInteger;
import javax.inject.Singleton;

@Singleton
public
class TestQueue extends Queue {

  public final AtomicInteger addCounter;
  public final CountDownLatch firstWriteLatch;

  public TestQueue() {
    super(Integer.MAX_VALUE);
    this.addCounter = new AtomicInteger();
    this.firstWriteLatch = new CountDownLatch(1);
  }

  @Override
  public boolean add(String value) {
    addCounter.incrementAndGet();
    firstWriteLatch.countDown();
    return super.add(value);
  }
}
